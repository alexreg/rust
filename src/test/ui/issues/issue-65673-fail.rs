#![feature(trait_alias)]

use std::any::Any;

trait Trait {}

trait WithType {
    type Ctx: ?Sized;
}

trait Alias<T> = Any where T: Trait;

// Note, we do not impl `Trait` for `()` here, so should expect the error below.

impl<T> WithType for T {
    type Ctx = dyn Alias<T>;
}

fn main() {
    let _: Box<<() as WithType>::Ctx> = Box::new(());
    //~^ ERROR the trait bound `(): Trait` is not satisfied [E0277]
}
