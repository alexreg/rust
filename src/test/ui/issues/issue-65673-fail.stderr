error[E0277]: the trait bound `(): Trait` is not satisfied
  --> $DIR/issue-65673-fail.rs:20:41
   |
LL |     let _: Box<<() as WithType>::Ctx> = Box::new(());
   |                                         ^^^^^^^^^^^^ the trait `Trait` is not implemented for `()`
   |
   = note: required for the cast to the object type `dyn Alias<()>`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0277`.
